src/front/glsl/variables.rs

```diff
@@ -0,0 +1,81 @@
+use crate::{
+    Binding, BuiltIn, Expression, GlobalVariable, Handle, ScalarKind, ShaderStage, StorageAccess,
+    StorageClass, Type, TypeInner, VectorSize,
+};
+
+use super::ast::*;
+
+pub fn lookup_variable(program: &mut Program, name: &str) -> Option<Handle<Expression>> {
```

Comment:
would it make sense to have it as a method on the `Program`? It can still be in a different module (like `variables` module here)

Reply:
Sure, is expecting this to grow large (lots of builtins), but if can still be in it's own file, that would be fine. I'll figure out how.